oracle/oracle.go

```diff
@@ -174,13 +211,26 @@ func (w *WorkOracle) Positions(start, end uint64) (WorkOracleResult, error) {
 		return WorkOracleResult{}, fmt.Errorf("invalid `start` and `end`; expected: start <= end, given: %v > %v", start, end)
 	}
 
-	res, err := w.scrypt.Positions(start, end)
-	if err != nil {
-		return WorkOracleResult{}, err
+	tries := 0
+	for {
+		res, err := w.scrypt.Positions(start, end)
+		tries += 1
+		switch err {
+		case postrs.ErrInitializationFailed:
+			w.options.logger.With().Warn("failure during initialization", zap.Error(err))
+			if tries < 1+w.options.maxRetries {
+				w.options.logger.With().Warn("retrying initialization", zap.Int("tries", tries))
+				time.Sleep(w.options.retryDelay)
+			} else {
+				return WorkOracleResult{}, fmt.Errorf("failed to initialize scrypt after %v tries", tries)
+			}
```

Comment:
Inverting allows to return early not requiring an `else` block:

```suggestion
			if tries > w.options.maxRetries {
				return WorkOracleResult{}, fmt.Errorf("failed to initialize scrypt after %v tries", tries)
			}
			w.options.logger.With().Warn("retrying initialization", zap.Int("tries", tries))
			time.Sleep(w.options.retryDelay)
```